crates/dojo-lang/src/plugin_test_data/system

```diff
@@ -84,6 +84,134 @@ mod withcomponent {
+
+#[dojo::interface]
+trait IFaultyTrait {
+    /// Even if it looks weird, the expected pattern for a constant in a trait is
+    /// `const <name>: <type>;`
+    /// See: https://github.com/starkware-libs/cairo/blob/600553d34a1af93ff7fb038ac677efdd3fd07010/crates/cairo-lang-parser/src/parser.rs#L835
+    const ONE: u8;
+
+    fn do_ref_self(ref self: TContractState);
+
+    #[my_attr]
+    fn do_with_attrs(p1: u8) -> u16;
+}
+
+#[dojo::interface]
+trait INominalTrait {
+    fn do_no_param();
+    fn do_no_param_but_self(self: @TContractState);
+    fn do_params(p1: dojo_examples::models::Direction, p2: u8);
+    fn do_params_and_self(self: @TContractState, p2: u8);
+    fn do_return_value(p1: u8) -> u16;
+}
+
+#[dojo::contract]
+mod MyFaultyContract {
+
+    fn do_with_ref_self(ref self: ContractState) -> felt252 {
+        'land'
+    }
+
+    fn do_with_several_world_dispatchers(
+        world: IWorldDispatcher,
+        vec: Vec2,
+        another_world: IWorldDispatcher
+    ) -> felt252 {
+        'land'
+    }
+
+    #[external(v0)]
+    impl ActionsImpl of IActions<ContractState> {
+        fn do_with_ref_self(ref self: ContractState) -> felt252 {
+            'land'
+        }
+
+        fn do_with_several_world_dispatchers(
+            world: IWorldDispatcher,
+            vec: Vec2,
+            another_world: IWorldDispatcher
+        ) -> felt252 {
+            'land'
+        }
+    }
+}
+
+#[dojo::contract]
+mod MyNominalContract {
+
+    #[derive(Drop)]
+    struct Action {
+        damage: u8
+    }
+
+    #[external(v0)]
+    #[computed]
+    fn do(vec: Vec2) -> felt252 {
+        'land'
+    }
+
+    #[external(v0)]
+    #[computed]
+    fn do_with_self(self: @ContractState, vec: Vec2) -> felt252 {
+        'land'
+    }
+
+    #[external(v0)]
+    #[computed]
+    fn do_with_world_first(world: IWorldDispatcher, vec: Vec2) -> felt252 {
+        'land'
+    }
+
+    #[external(v0)]
+    #[computed]
+    fn do_with_self_and_world_first(self: @ContractState, world: IWorldDispatcher, vec: Vec2) -> felt252 {
+        'land'
+    }
+
+    #[external(v0)]
+    #[computed]
+    fn do_with_world_not_first(vec: Vec2, world: IWorldDispatcher) -> felt252 {
+        'land'
+    }
+
+    #[external(v0)]
+    #[computed]
+    fn do_with_self_and_world_not_first(self: @ContractState, vec: Vec2, world: IWorldDispatcher) -> felt252 {
+        'land'
+    }
+
+    #[external(v0)]
+    impl ActionsImpl of IActions<ContractState> {
+
+        fn do(vec: Vec2) -> felt252 {
+            'land'
+        }
+
+        fn do_with_self(self: @ContractState, vec: Vec2) -> felt252 {
+            'land'
+        }
+
+        fn do_with_world_first(world: IWorldDispatcher, vec: Vec2) -> felt252 {
+            'land'
+        }
+
+        fn do_with_self_and_world_first(self: @ContractState, world: IWorldDispatcher, vec: Vec2) -> felt252 {
+            'land'
+        }
+
+        fn do_with_world_not_first(vec: Vec2, world: IWorldDispatcher) -> felt252 {
```

Comment:
Isn't this faulty? :P
(and the one below)

Well.. it depends if the current impl is only considering the `world` as first position to be replaced, and the other ones to be kept.
But as mentioned earlier, and @tarrencev correct me if I'm wrong on that, we always want the `world: IWorldDispatcher` to be inlined, hence faulty IMHO.

Reply:
If we decide to force `world` to be the first parameter, yes, this is faulty 👍